### PR TITLE
AwsSdkClient refactor

### PR DESCRIPTION
**Pull request by abrooksv**
* Introduce an interface for it so it can be mocked easier
* Remove the suppliers from the profile factory for the http client

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hunterwerlla** [comment]
What's the difference between `sharedSdkClient` and `sdkHttpClient`?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt [hunk 1]
```diff
@@ -190,7 +190,7 @@ class ProfileCredentialProviderFactory(private val ssoCache: SsoCache = diskCach
 
         // Some profiles failed to load
         if (newProfiles.invalidProfiles.isNotEmpty()) {
-            val message = newProfiles.invalidProfiles.values.joinToString("\n") { it.message ?: it::class.java.name }
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt:193]: What does this change?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt:193]: Whoops, shouldn't be in this PR

**Comment by abrooksv** [comment]
> What's the difference between `sharedSdkClient` and `sdkHttpClient`?

None, function vs val.

Found the shared name a bit clearer that we share it instead of making a new one

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**